ssl/tls13_enc.c

```diff
@@ -440,18 +470,19 @@ int tls13_change_cipher_state(SSL_CONNECTION *s, int which)
     const unsigned char *label;
     size_t labellen, hashlen = 0;
     int ret = 0;
-    const EVP_MD *md = NULL;
+    const EVP_MD *md = NULL, *mac_md = NULL;
     const EVP_CIPHER *cipher = NULL;
+    int mac_pkey_type = NID_undef;
     SSL_CTX *sctx = SSL_CONNECTION_GET_CTX(s);
-    size_t keylen, ivlen, taglen;
+    size_t keylen, ivlen = EVP_MAX_IV_LENGTH, taglen;
+    EVP_MD_CTX *mdctx = NULL;
     int level;
     int direction = (which & SSL3_CC_READ) != 0 ? OSSL_RECORD_DIRECTION_READ
                                                 : OSSL_RECORD_DIRECTION_WRITE;
 
     if (((which & SSL3_CC_CLIENT) && (which & SSL3_CC_WRITE))
             || ((which & SSL3_CC_SERVER) && (which & SSL3_CC_READ))) {
         if (which & SSL3_CC_EARLY) {
-            EVP_MD_CTX *mdctx = NULL;
```

Comment:
Not sure I understand why the scope of this was changed?

Reply:
It was done to remove the multiple call to `EVP_MD_CTX_free(mdctx)`. But your suggestion regarding relocation of creation of `mdctx` looks better.